pkg/restapi/vc/operation/operations.go

```diff
@@ -642,3 +674,18 @@ func (o *Operation) parseAndVerifyVC(vcBytes []byte) (*verifiable.Credential, er
 
 	return vc, nil
 }
+
+func (o *Operation) parseAndVerifyVP(vpBytes []byte) (*verifiable.Presentation, error) {
+	vp, err := verifiable.NewPresentation(
+		vpBytes,
+		verifiable.WithPresEmbeddedSignatureSuites(ed25519signature2018.New()),
+		verifiable.WithPresPublicKeyFetcher(
+			verifiable.NewDIDKeyResolver(o.vdri).PublicKeyFetcher(),
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return vp, nil
```

Comment:
Should you check credential status here (call verifyVC)?

Reply:
added that

Reply:
looks good